hyperspy/axes.py

```diff
@@ -668,6 +633,483 @@ def update_from(self, axis, attributes=["scale", "offset", "units"]):
+            exist in either of the AxesManagers, an AttributeError will be
+            raised. If None, the parameters of `expression` are updated.
+        Returns
+        -------
+        A boolean indicating whether any changes were made.
+
+        """
+        if attributes is None:
+            attributes = self.parameters_list
+        return super().update_from(axis, attributes)
+
+    def get_axis_dictionary(self):
+        d = super().get_axis_dictionary()
+        d['expression'] = self._expression
+        d.update({'size': self.size, })
+        d.update({'x': self.x.get_axis_dictionary(), })
+        for kwarg in self.parameters_list:
+            d[kwarg] = getattr(self, kwarg)
+        return d
+
+    def convert_to_non_uniform_axis(self):
+        d = super().get_axis_dictionary()
+        d["_type"] = 'DataAxis'
+        self.__class__ = DataAxis
+        self.__init__(**d, axis=self.axis)
+        del self._expression
+        del self._function
+        self.remove_trait('x')
+
+    def crop(self, start=None, end=None):
+        """Crop the axis in place.
+
+        Parameters
+        ----------
+        start : int, float, or None
+            The beginning of the cropping interval. If type is ``int``,
+            the value is taken as the axis index. If type is ``float`` the index
+            is calculated using the axis calibration. If `start`/`end` is
+            ``None`` the method crops from/to the low/high end of the axis.
+        end : int, float, or None
+            The end of the cropping interval. If type is ``int``,
+            the value is taken as the axis index. If type is ``float`` the index
+            is calculated using the axis calibration. If `start`/`end` is
+            ``None`` the method crops from/to the low/high end of the axis.
+
+        Note
+        ----
+        When cropping an axis with descending axis values based on the axis
+        calibration, the `start`/`end` tuple also has to be in descending
+        order.
+        """
+
+        slice_ = self._get_array_slices(slice(start, end))
+        self.x.crop(start=slice_.start, end=slice_.stop)
+        self.size = self.x.size
+        self.update_axis()
+    crop.__doc__ = DataAxis.crop.__doc__
+
+    def _slice_me(self, slice_):
+        """Returns a slice to slice the corresponding data axis and
+        change the offset and scale of the UniformDataAxis accordingly.
+
+        Parameters
+        ----------
+        slice_ : {float, int, slice}
+
+        Returns
+        -------
+        my_slice : slice
+
+        """
+        my_slice = self._get_array_slices(slice_)
+        self.x._slice_me(my_slice)
+        self.update_axis()
+        return my_slice
+
+
+class UniformDataAxis(BaseDataAxis, UnitConversion):
+    def __init__(self,
+                 index_in_array=None,
+                 name=t.Undefined,
+                 units=t.Undefined,
+                 navigate=t.Undefined,
+                 size=1.,
```

Comment:
The size parameter should be an int, not float.

Reply:
see #2693